feat(setup): SCRUM-2 Create Symfony project with Docker and quality tools setup

adnlv opened the pull request:
Jira: [SCRUM-2](https://epam-team-php.atlassian.net/browse/SCRUM-2)

# What this PR does and Why?

- Create a Symfony project, using [symfony-docker](https://github.com/dunglas/symfony-docker) as a reference. It uses [FrankenPHP](https://frankenphp.dev) with MySQL in Docker containers and code quality tools, such as `phpstan` and `php-cs-fixer`.
- Provide a simple GitHub Actions CI workflow.

CPAprince committed:
Update README.md

Branch & Jira rules

adnlv committed:
Add a basic Docker setup

adnlv committed:
Try to resolve CI problem by adding necessary directories

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
-# Twitter
-Lab's Team Project 
+#### --== Twitter ==-- ####
```
```suggestion
# Twitter
```

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
-Lab's Team Project 
+#### --== Twitter ==-- ####
+
+Lab's Team Project
```
@adanyl0v I bet you can come up with a better description.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+Lab's Team Project
+
+# 📘 Правила роботи з Git у проєкті
```
@adanyl0v It's ok for the course to use Ukrainian, but English is better as it's the main language of IT and programming.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+# 📘 Правила роботи з Git у проєкті
+
+У цьому проєкті використовується безпечна та зрозуміла модель розробки з чітким розмежуванням гілок:
```
@adanyl0v The goal of the `README` in the root of the project is to explain the quick start. Other documentation, like git branching, etc. can be placed in a separate folder `/docs` and set in the readme as a link.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 🌿 Структура гілок
```
@adanyl0v From line 15 to line 98, you can save space by keeping it simple. Reuse existing branching strategy like [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/). Share the link. No need to explain every detail.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 📝 Правила комітів
```
@adanyl0v The same issues. Too many texts. The way to format commit messages is clear, but it can also be reused from existing commit conventions, like [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). Replace with the link.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 🚫 Заборонено
```
@adanyl0v This one is a good thing to mention.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 🌿 Структура гілок
```
@adanyl0v Partially duplicates content from line 15 to line 55.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 🎫 Інтеграція з Jira
```
@adanyl0v It's not quite correct to call it an integration, as you don't have any integration yet. This one is correct, but too much text is used to describe a thing which is a standard for most projects. Keep it as a part of the commit message guide.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## ✅ End of Sprint / Release
```
@adanyl0v You can remove all the content below because:
1. It doesn't belong here (Jira statuses in Github?)
2. Duplicated content (Заборонено?Схема потоку?)

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+### 3. Push + Pull Request
```
@adanyl0v Options 3 & 4 are related to PR creation. Better to replace with a template. Check [github guide](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) and replace this with a template.

ihpr commented on README.md:
```diff
@@ -1,2 +1,336 @@
+
+---
+
+## 🚀 Процес роботи в Scrum
```
@adanyl0v The GitHub documentation should store only code-specific docs or repository-specific ones. Jira staff is not like that. Move it to another documentation store. Better places: Confluence, Notion, others.

ihpr commented on src/Controller/HealthController.php:
```diff
@@ -0,0 +1,18 @@
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class HealthController extends AbstractController
```
@adanyl0v Better to use a [controller as a service](https://symfony.com/doc/current/controller/service.html) in Symfony to avoid any inheritance of a vendor-based controller.

ihpr replied:
@adanyl0v Final - Like.

ihpr commented on src/Controller/HealthController.php:
```diff
@@ -0,0 +1,18 @@
+<?php
+
```
@adanyl0v Use `declare(strict_types=1);`.

ihpr commented on src/DataFixtures/AppFixtures.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
@adanyl0v Remove this example as a rudiment.

ihpr commented on config/packages/doctrine.yaml:
```diff
@@ -0,0 +1,48 @@
+        auto_mapping: true
+        mappings:
+            App:
+                type: attribute
```
@adanyl0v Use [XML mapping](https://www.doctrine-project.org/projects/doctrine-orm/en/3.5/reference/xml-mapping.html) instead of attributes.

ihpr commented on config/packages/doctrine.yaml:
```diff
@@ -0,0 +1,48 @@
+            App:
+                type: attribute
+                is_bundle: false
+                dir: '%kernel.project_dir%/src/Entity'
```
@adanyl0v What fi we are going to use the module approach for design? 1 entity per 1 model.
Here is the structure:
```
src
- Auth (Entity/UserEntity)
- Tweet (Model/TweetEntity)
- Analytics (*/Model/*)
```

Update it.

ihpr commented on config/packages/doctrine.yaml:
```diff
@@ -0,0 +1,48 @@
+                is_bundle: false
+                dir: '%kernel.project_dir%/src/Entity'
+                prefix: 'App\Entity'
+                alias: App
```
@adanyl0v App is too common. Every application is an App. Does your app make a difference?

ihpr commented on config/packages/doctrine.yaml:
```diff
@@ -0,0 +1,48 @@
+        validate_xml_mapping: true
+        naming_strategy: doctrine.orm.naming_strategy.underscore_number_aware
+        identity_generation_preferences:
+            Doctrine\DBAL\Platforms\PostgreSQLPlatform: identity
```
@adanyl0v PostgreSQL? I thought we are going to use MySQL?

ihpr commented on config/packages/framework.yaml:
```diff
@@ -0,0 +1,15 @@
+    secret: '%env(APP_SECRET)%'
+
+    # Note that the session will be started ONLY if you read or write from it.
+    session: true
```
@adanyl0v REST API is our main architectural pattern. It's stateless. No sessions will be used.

ihpr commented on config/packages/mercure.yaml:
```diff
@@ -0,0 +1,8 @@
+mercure:
```
@adanyl0v We are not going to use it for now. Please remove that.

ihpr commented on config/routes.yaml:
```diff
@@ -0,0 +1,11 @@
+# To list all registered routes, run the following command:
+#   bin/console debug:router
+
+controllers:
```
@adanyl0v Check this config to match the module approach.

```
src
- Module
- - UI/Interface/Presentation
- - - (CLI, Web, REST, gRPC)
- - - - (Command, Controller)
```

ihpr commented on config/bundles.php:
```diff
@@ -0,0 +1,10 @@
+    Symfony\Bundle\FrameworkBundle\FrameworkBundle::class => ['all' => true],
+    Doctrine\Bundle\DoctrineBundle\DoctrineBundle::class => ['all' => true],
+    Doctrine\Bundle\MigrationsBundle\DoctrineMigrationsBundle::class => ['all' => true],
+    Symfony\Bundle\MercureBundle\MercureBundle::class => ['all' => true],
```
@adanyl0v Remove mercury. Not needed for now.

ihpr commented on config/bundles.php:
```diff
@@ -0,0 +1,10 @@
+    Doctrine\Bundle\MigrationsBundle\DoctrineMigrationsBundle::class => ['all' => true],
+    Symfony\Bundle\MercureBundle\MercureBundle::class => ['all' => true],
+    Doctrine\Bundle\FixturesBundle\DoctrineFixturesBundle::class => ['dev' => true, 'test' => true],
+    Symfony\Bundle\MakerBundle\MakerBundle::class => ['dev' => true],
```
@adanyl0v Forget about this. Remove.

ihpr commented on docs/setup.md:
```diff
@@ -0,0 +1,25 @@
+# Symfony Franken MySQL API Template
+
+## Getting started
```
@adanyl0v Getting started must be in `README`.

ihpr commented on .php-cs-fixer.dist.php:
```diff
@@ -0,0 +1,13 @@
+
+return (new PhpCsFixer\Config())
+    ->setRules([
+        '@Symfony' => true,
```
@adanyl0v How about https://cs.symfony.com/doc/rules/strict/declare_strict_types.html?

ihpr commented on phpstan.dist.neon:
```diff
@@ -0,0 +1,8 @@
+parameters:
+    level: 6
```
@adanyl0v Why not 4? Why not 1?

adnlv committed:
fix(mercure): SCRUM-2 Remove Mercure

- Remove `symfony/mercure-bundle bundle and clear out related includes

- Expose database port for public usage

adnlv committed:
fix(maker): SCRUM-2 Remove maker-bundle

- Get rid of the `symfony/maker-bundle` as we are not experienced enough
 to use it properly

 - Set PHPStan analysis level to 4

 - Remove a redundant fixtures example

adnlv committed:
feat(quality): SCRUM-2 Use additional `php-cs-fixer` strict rules

- Use `strict_comparison` and `declare_strict_types`
  rules for better code quality

- Rename project namespace to `Twitter`

- Remove `HealthController` dependency from `AbstractController`.
  Only the `GET` method is allowed. Append `/api` prefix to the path

adnlv committed:
feat(doctrine): SCRUM-2 Use XML mapping for Doctrine entities

- Prepare Doctrine configuration for the modular architecture design by
  mapping entities in corresponding `.orm.xml` files

- Register controllers manually

adnlv committed:
feat(doctrine): SCRUM-2 Disable sessions in `framework.yaml` since we won't use them

adnlv committed:
fix(healthCheck): SCRUM-2 Refactor health check module

adnlv committed:
feat(migration): SCRUM-2 Generate a migration for `HealthCheckReport`

adnlv committed:
docs: SCRUM-2 Add controller registration and entity mapping guides

- Create a controller creation and registration documentation

- Create a Doctrine entity XML mapping documentation.
  Provide console commands generating and validating migrations.

adnlv committed:
docs: SCRUM-2 Reduce unnecessary bloat in `README.md`

Keep `README.md` as simple as possible by moving Git workflow
rules into a separate file. Create a pull request template.

adnlv committed:
feat: SCRUM-2 Add `php` container health check and improve Caddy config

ihpr commented on src/HealthCheck/Entity/HealthCheckReport.php:
```diff
@@ -0,0 +1,26 @@
+
+namespace Twitter\HealthCheck\Entity;
+
+final class HealthCheckReport
```
@adanyl0v Move to doc.

ihpr commented on config/packages/cache.yaml:
```diff
@@ -0,0 +1,19 @@
+framework:
+    cache:
+        # Unique name of your app: used to compute stable namespaces for cache keys.
```
@adanyl0v Rudiment comments

adnlv committed:
style(config): SCRUM-2 Remove rudiment comments to simplify configs

adnlv committed:
refactor(doctrine): SCRUM-2 Add an XML mapping example to the documentation

Entities in `HealthCheck` module serve no real purpose, it is better
to add them to `docs` as an XML mapping example.
